src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java

```diff
@@ -26,13 +26,47 @@ public String getOperator() {
     }
   }
 
+  // TODO: why are these not final ?
+  public List<LogicalExpression> logicalExpressions;
+  public List<ComparisonExpression> comparisonExpressions;
+
   private LogicalOperator logicalRelation;
+  // TODO - there MUST be explaination of why this count is needed, why not just use the size of the
+  // list ?
+  // Also, the list is public so anyone can change it and break this count.
   private int totalComparisonExpressionCount;
   private int totalIdComparisonExpressionCount;
 
+  // TODO: All the uses of this ctor only pass a different opertor, all other args are the same
+  private LogicalExpression(
+      LogicalOperator logicalRelation,
+      int totalComparisonExpressionCount,
+      List<LogicalExpression> logicalExpressions,
+      List<ComparisonExpression> comparisonExpression) {
+    this.logicalRelation = logicalRelation;
+    this.totalComparisonExpressionCount = totalComparisonExpressionCount;
+    this.logicalExpressions = logicalExpressions;
+    this.comparisonExpressions = comparisonExpression;
+  }
+
+  public static LogicalExpression and() {
+    return new LogicalExpression(LogicalOperator.AND, 0, new ArrayList<>(), new ArrayList<>());
+  }
+
+  public static LogicalExpression or() {
+    return new LogicalExpression(LogicalOperator.OR, 0, new ArrayList<>(), new ArrayList<>());
+  }
+
+  public static LogicalExpression not() {
+    return new LogicalExpression(LogicalOperator.NOT, 0, new ArrayList<>(), new ArrayList<>());
+  }
+
   /** This method will flip the operators and operand if logical operator is not */
+  // TODO: comment to explain what is going on here
```

Comment:
Above comment kind of does, could maybe elarobate and indicate "not" is operator in the sentence.

Reply:
Yuqi Resolve: added comment